### DISCUSSION TITLE
Does SignalDB require the entire collection to be in memory?

### DISCUSSION THREAD

**Discussion by beorn** [maxnowack/signaldb#1008] [Q&A] [answered]
The SignalDB docs talk about SignalDB being an in-memory database, but I see that it has the concept [memory adapters](https://signaldb.js.org/core-concepts/#memory-adapters).  Would it be feasible to have the entire collection be backed by a sqlite/pglite database?

Is async collections the bottleneck?  #397 

Also, since the [AutoFetchReplication](https://signaldb.js.org/replication/#autofetchcollection) has been deprecated in favor of SyncManager, how can you do on-demand fetching?

**Comment by maxnowack** [comment, marked as answer]
For the most use cases, it should be fine to have all data in memory. However, if you need to handle a lot of data, memory could be insufficient to handle these amounts. That's exactly what I want to tackle with #397 but I'm not yet 100% sure how I could implement it.
Currently it is possible with the ´AutoFetchCollection´ to on-demand fetch data from a sqlite/pglite database. I always thought about fetching remote data with the `AutoFetchCollection` but It's actually also quite useful for fetching asynchronous storage.
